copycat/runtime/src/main/java/org/apache/kafka/copycat/runtime/WorkerSinkTask.java

```diff
@@ -171,12 +185,8 @@ public void poll(long timeoutMs) {
      **/
     public void commitOffsets(boolean sync, final int seqno) {
         log.info("{} Committing offsets", this);
-        final HashMap<TopicPartition, OffsetAndMetadata> offsets = new HashMap<>();
-        for (TopicPartition tp : consumer.assignment()) {
-            long pos = consumer.position(tp);
-            offsets.put(tp, new OffsetAndMetadata(pos));
-            log.debug("{} committing {} offset {}", id, tp, pos);
-        }
+
+        final Map<TopicPartition, OffsetAndMetadata> offsets = new HashMap<>(currentOffsets);
```

Comment:
In general we need copies for anything we hand to the user to ensure anything they do will not affect the rest of our code. Here if the `flush` call does anything to modify the map, the following offset commit will be incorrect. (Previously, we also needed to because earlier versions of KafkaConsumer were not defensively copying the offsets used during async commit, but that has since been fixed.)